### PR TITLE
Set node config for pods without dedicated instance configs

### PR DESCRIPTION
**Pull request by puppetninja**
This is a duplicated one of PR #500 , generated from a branch in official repo for all the CI checks to be run

In combination with a horizontal pod autoscaler (not supported in tezos-k8s yet), this will ensure that any additional replicas spun up by the autoscaler will also be configured according to the statefulset-level config in values.yaml (and node_globals as well)

### CODE REVIEW

**Review by puppetninja — COMMENTED** [review]
> this will cause issue when there are multiple nodes classes specified in nodes as we cannot know which class the pod belongs to

Are you sure? I just tried with the following config:

```
 archive-baking-node:                                                                                                                                                               
    runs:                                                                                                                                                                            
      - octez_node                                                                                                                                                                   
      - baker                                                                                                                                                                        
    storage_size: 15Gi                                                                                                                                                               
    config:                                                                                                                                                                          
      shell:                                                                                                                                                                         
        history_mode: archive                                                                                                                                                        
      metrics_addr:                                                                                                                                                                  
        - :9932                                                                                                                                                                      
    instances:                                                                                                                                                                       
      - is_bootstrap_node: true                                                                                                                                                      
        bake_using_accounts:                                                                                                                                                         
          - archive-baking-node-0                                                                                                                                                    
  archive2-baking-node:                                                                                                                                                              
    runs:                                                                                                                                                                            
      - octez_node                                                                                                                                                                   
    storage_size: 15Gi                                                                                                                                                               
    config:                                                                                                                                                                          
      shell:                                                                                                                                                                         
        history_mode: archive                                                                                                                                                        
      metrics_addr:                                                                                                                                                                  
        - :9933                                                                                                                                                                      
    instances:                                                                                                                                                                       
      - is_bootstrap_node: true                                       
```

And I scaled up `archive2-baking-node` to 2, the new pod came up and had a `MY_NODE_CLASS` env variable set to `archive2-baking-node`. It had the configuration I was expecting.

**Comment by harryttd** [comment]
Try scaling up the first class. I think you will see it has the wrong class name. I will leave a comment inline.

**Review by harryttd — DISMISSED** [review]

File: utils/config-generator.py [hunk 1]
```diff
@@ -33,11 +33,11 @@
 
 for cl, val in NODES.items():
     if val != None:
+        MY_POD_CLASS = val
```
[the remark this answers is not in the document]
Comment (puppetninja) [reply utils/config-generator.py]: Thanks @harryttd I have updated the commit based on your comment, can you have a double check ?

**Review by nicolasochem — APPROVED** [review]
LGTM, it does not affect whichever scenario we currently support (because MY_POD_CLASS is always set in this case) and appears to make autoscaling work.

**Pull request merged**